feat: Add Wordpress cataloger

disc opened the pull request:
Add a cataloger that detects installed Wordpress plugins by looking files in `*/wp-content/plugins/*` directories
Closes #1911 

Related MR to [Grype](https://github.com/anchore/grype)  for removing `wordpress` from `known` targets and unsuppressing of wordpress plugins vulnerabilities
https://github.com/anchore/grype/pull/1553

wagoodman commented on syft/pkg/cataloger/wordpress/parse_plugin.go:
```diff
@@ -0,0 +1,94 @@
+	var pkgs []pkg.Package
+	var fields = make(map[string]interface{})
+
+	bytes, err := io.ReadAll(reader)
```
we're really only expecting to parse the top comment of the file. I think we should try a best-effort at not searching the entire file, but instead to the first `*/` or only the first 100 lines. 

That's my initial take since this is capturing on potentially several PHP files. What do you think @disc ?

disc replied:
Hey, I agree. Will read first N kilobytes (for example, 4kb) into a buffer instead.

wagoodman reviewed (commented):
@disc I've caught up this branch based on the several changes made on main recently. I have a couple of small questions, otherwise I feel this is about the cross the finish line 🙌

wagoodman reviewed (dismissed):
Thanks for adding this cataloger 🙌 !  (and thanks for your patience on this PR -- it came in the midst of an API refactor so needed to land afterwards).

wagoodman commented:
A couple of clarifying questions:
```json
{
  "id": "62595455b80ed7ff",
  "name": "Akismet Anti-spam: Spam Protection",
  "version": "5.3",
  "type": "wordpress-plugin",
  "foundBy": "wordpress-plugin-cataloger",
  ...  
  "metadata": {
    "plugin_name": "akismet",
    "author": "Automattic - Anti-spam Team",
    "author_uri": "https://automattic.com/wordpress-plugins/"
  }
}

```

1. Should the `pkg.Package.Name` be what is currently the `pkg.WordpressPlugin.PluginName`? (I could reason either way, curious for opinions here)
2. (Completely independent from the first question) Should `pkg.WordpressPlugin.PluginName` really be something like `pkg.WordpressPlugin.PluginInstallDirectory` (or similar) since it's derived from an installation path?

disc commented:
> @disc A couple of clarifying questions:
> 
> ```json
> {
>   "id": "62595455b80ed7ff",
>   "name": "Akismet Anti-spam: Spam Protection",
>   "version": "5.3",
>   "type": "wordpress-plugin",
>   "foundBy": "wordpress-plugin-cataloger",
>   ...  
>   "metadata": {
>     "plugin_name": "akismet",
>     "author": "Automattic - Anti-spam Team",
>     "author_uri": "https://automattic.com/wordpress-plugins/"
>   }
> }
> ```
> 
>     1. Should the `pkg.Package.Name` be what is currently the `pkg.WordpressPlugin.PluginName`? (I could reason either way, curious for opinions here)
> 
>     2. (Completely independent from the first question) Should `pkg.WordpressPlugin.PluginName` really be something like `pkg.WordpressPlugin.PluginInstallDirectory` (or similar) since it's derived from an installation path?
> 
> 
> (I can make the updates if there are any changes needed)

Hey @wagoodman. Looks like I need to make a few changes after a huge refactor and probably these questions won't be actual further.

wagoodman commented:
I can help with restoring/regenerating the JSON schema if needed. Also, is the refactor you're referring to one that happened on main (touching the cataloging API)? Or are you talking about a refactor within the new wordpress plugin cataloger you wrote?

disc commented:
> is the refactor you're referring to one that happened on main (touching the cataloging API)?

Yes, I meant this one. I've just completed all the needed changes - now need to look into a test run.

wagoodman commented:
Hey, I fixed the failing tests and updated the branch. I still have some questions left (from https://github.com/anchore/syft/pull/2218#issuecomment-1924531140)
 
1. Should the value in `pkg.Package.Name` really be what is currently in the `pkg.WordpressPlugin.PluginName` field? Said another way, should the package name be the install directory name instead of the plugin display name? 
1. Should `pkg.WordpressPlugin.PluginName` really be renamed to `pkg.WordpressPlugin.PluginInstallDirectory` (or similar) since it's derived from an installation path and not actually the `Plugin Name:` field from the underlying source code?

disc commented:
> Hey, I fixed the failing tests and updated the branch. I still have some questions left (from [#2218 (comment)](https://github.com/anchore/syft/pull/2218#issuecomment-1924531140))
> 
>     1. Should the value in `pkg.Package.Name` really be what is currently in the `pkg.WordpressPlugin.PluginName` field? Said another way, should the package name be the install directory name instead of the plugin display name?
> 
>     2. Should `pkg.WordpressPlugin.PluginName` really be renamed to `pkg.WordpressPlugin.PluginInstallDirectory` (or similar) since it's derived from an installation path and not actually the `Plugin Name:` field from the underlying source code?

1) I believe yes, it makes the package name short and simple, sometimes it contains long names such as `Akismet Anti-spam: Spam Protection`
2) Probably yes, it's contains almost the same as a name but uses for building CPEs

wagoodman commented:
It sounds like both answers point to making some small updates -- I can take a stab at it if you'd like? (or if you wanted to that's also alright)

wagoodman reviewed (approved):
I force pushed a rebase (there were a few commits from main that didn't belong). I also regenerated the JSON schema after changing the field caps to be consistent with the field convention we use (camel case).

Overall great work -- thanks for taking the time to add this cataloger 🙏 !
